metaswitch/common/generate_stats_csv.py

```diff
@@ -0,0 +1,266 @@
+# * Enhance `merge_entry` to merge the value to the right place
+#   in the output CSV.
+# * Enhance `COLUMN_HEADERS` to output the new field.
+import os
+import argparse
+import logging
+import csv
+import StringIO
+import collections
+import mib
+
+logger = logging.getLogger(__name__)
+
+COLUMN_HEADERS = [
+    "Version Introduced",
+    "MIB Table Name",
+    "MIB Field Name",
+    "Source File",
+    "MIB Table Description",
+    "OID",
+    "MIB Field Description",
+    "Calculation Type",
+    "Reset Trigger",
+    "Reset Trigger Detail",
+    "Data Type",
+    "Index Fields",
+    "Index Field Values",
+    "Field Size",
+    "Field Units",
+    "Aggregation Mechanism",
+]
+
+
+def merge_entry(key, mib_data, csv_data):
+    """Create a tuple in the order specified by COLUMN_HEADERS.
+
+    Inputs are KeyData, MibData and CsvData respectively.
+    """
+    return (
+        csv_data.version_introduced,
+        key.mib_table_name,
+        key.mib_field_name,
+        mib_data.source_file,
+        mib_data.mib_table_description,
+        mib_data.oid,
+        mib_data.mib_field_description,
+        csv_data.calculation_type,
+        csv_data.reset_trigger,
+        csv_data.reset_trigger_detail,
+        csv_data.data_type,
+        csv_data.index_fields,
+        csv_data.index_field_values,
+        csv_data.field_size,
+        csv_data.field_units,
+        csv_data.aggregation_mechanism,
+    )
+
+MibData = collections.namedtuple("MibData", ["source_file",
+                                             "mib_table_description",
+                                             "mib_field_description",
+                                             "oid"])
+CsvData = collections.namedtuple("CsvData", ["version_introduced",
+                                             "calculation_type",
+                                             "reset_trigger",
+                                             "reset_trigger_detail",
+                                             "data_type",
+                                             "index_fields",
+                                             "index_field_values",
+                                             "field_size",
+                                             "field_units",
+                                             "aggregation_mechanism"])
+KeyData = collections.namedtuple("KeyData", ["mib_table_name",
+                                             "mib_field_name"])
+
+
+def main():
+    """Main entry point for the script."""
+    setup_logging()
+    args = parse_args()
+
+    parsed_mibs = parse_mib_files(args['mib_files'])
+    parsed_csv = parse_csv_file(args['csv_file'])
+
+    merged_entries = merge_csv_with_mibs(parsed_csv, parsed_mibs)
+
+    write_csv(merged_entries, args['output_file'])
+
+
+def setup_logging():
+    """Set up basic logging."""
+    level = os.getenv('LOG_LEVEL', 'WARNING')
+    logging.basicConfig(level=getattr(logging, level))

```

Comment:
Feels like we're mixing up logging and user feedback, Currently logging is sent to screen alongside user feedback. Don't necessarily need to do this now but generally I'd expect all logs to go to disk and the warning/error level logs to go to screen.